src/main/java/app/linguistai/bmvp/service/profile/ProfileService.java

```diff
@@ -73,7 +96,41 @@ public RUserProfile updateUserProfile(String email, QUserProfile profile) throws
             }
 
             List<String> userHobbies = hobbyService.updateUserHobby(dbUser, profile.getHobbies());
-            return new RUserProfile(dbUser.getId(), dbProfile.getName(), dbProfile.getBirhtDate(), dbProfile.getEnglishLevel(), userHobbies);
+
+            RUserProfile userProfile = RUserProfile.builder()
+                .id(dbUser.getId())
+                .name(dbProfile.getName())
+                .birhtDate(dbProfile.getBirhtDate())
+                .englishLevel(dbProfile.getEnglishLevel())
+                .hobbies(userHobbies)
+                .build();
+
+            if (ML_SERVICE_BASE_URL == null || ML_SERVICE_BASE_URL.isEmpty()) {
+                logger.warn("ML Service Base URL is not set!");
+                return userProfile;
+            }
+
+            // Create request body for the request to ML service
+            Map<String, Object> requestBody = new HashMap<>();
+            requestBody.put("profile", userProfile);
+
+            getWebClient().put()
+                .uri(ServiceUris.ML_SERVICE_UPDATE_PROFILE)
+                .header(Header.USER_EMAIL, email)
+                .body(Mono.just(requestBody), Map.class)
+                .retrieve()
+                .bodyToMono(String.class)
+                .subscribe(response -> {
+                    if (response != null) {
+                        logger.info("updateProfile - Response from ML service: " + response);
+                    }
+                }, error -> {
+                    if (error != null) {
+                        logger.error("updateProfile - Error from ML service: " + error.getMessage());
```

Comment:
Make sure to change the logging here when you remove slf4j

Reply:
Applied